### PR TITLE
build all of Katello on EL9

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: packages/katello/rubygem-katello/rubygem-katello.spec [hunk 1]
```diff
@@ -170,6 +172,9 @@ done
 %{foreman_plugin_log}
 
 %changelog
+* Mon Feb 12 2024 Evgeni Golov - 4.12.0-0.3.pre.master
+- Correct (Build)Requiremens for EL9
```
Comment (ekohl) [inline packages/katello/rubygem-katello/rubygem-katello.spec:176]:
```suggestion
- Correct (Build)Requirements for EL9
```

File: packages/katello/rubygem-katello/rubygem-katello.spec [hunk 2]
```diff
@@ -45,6 +45,8 @@ BuildRequires: rubygem(apipie-rails) >= 0.5.14
 BuildRequires: rubygem(fx) < 1.0
 BuildRequires: rubygem(pg)
 BuildRequires: rubygem(anemone)
+BuildRequires: rubygem(webrick)
+Requires: rubygem(webrick)
```
Comment (ekohl) [inline packages/katello/rubygem-katello/rubygem-katello.spec:49]: Since this is only to patch anemone, I'd prefer to do it there. Also, this section will get overwritten by our auto-update so that would revert the changes. So if we keep it, it should be outside of the generated section.

Comment (evgeni) [reply packages/katello/rubygem-katello/rubygem-katello.spec:49]: This was "intentional" so that the changes from https://github.com/Katello/katello/pull/10885 fix it "correctly" (that PR adds `webrick` as otherwise CI fails with the same error)

Comment (evgeni) [reply packages/katello/rubygem-katello/rubygem-katello.spec:49]: Opened https://github.com/theforeman/foreman-packaging/pull/10411 to fix anemone directly

**Comment by evgeni** [comment]
Repoclosure is related, needs https://github.com/theforeman/foreman-packaging/pull/10412 which needs Candlepin on EL9

File: packages/katello/rubygem-katello/rubygem-katello.spec [hunk 3]
```diff
@@ -25,9 +25,9 @@ Requires: foreman >= %{foreman_min_version}
 BuildRequires: foreman-assets >= %{foreman_min_version}
 BuildRequires: foreman-plugin >= %{foreman_min_version}
 Requires: ruby >= 2.5
-Requires: ruby < 3
+Requires: ruby < 3.1
 BuildRequires: ruby >= 2.5
-BuildRequires: ruby < 3
+BuildRequires: ruby < 3.1
```
Comment (ekohl) [inline packages/katello/rubygem-katello/rubygem-katello.spec:30]: I suspect we'll raise this to >= 2.7 and < 4.

Comment (evgeni) [reply packages/katello/rubygem-katello/rubygem-katello.spec:30]: Probably. My goal was "update it to something so it builds, and then when the actual Ruby3 PR is in, we regenerate the deps anyway"

**Comment by evgeni** [comment]
Should be merged together with https://github.com/theforeman/foreman-packaging/pull/10412 as that contains `katello` which `requires: rubygem-katello`

**Review by ekohl — APPROVED** [review]
Will merging this without support in katello master also break the nightly package build?

At least 👍 for the changes once the core PR is merged

**Comment by evgeni** [comment]
> Will merging this without support in katello master also break the nightly package build?
> 
> At least 👍 for the changes once the core PR is merged

That's the hackish beauty of it: As you can see in this PR's test results, things build on EL9 with Ruby 3 and NodeJS 18 just fine.
Several parts of the app are broken (and that's what https://github.com/Katello/katello/pull/10885 is fixing), but it's sufficiently OK to build and produce RPM that can then be tested.
So no, merging this will not break anything, but allow us to produce nightlies that can then be further tested.
